crates/bevy_sprite/src/render/mod.rs

```diff
@@ -389,11 +400,13 @@ pub fn queue_sprites(
         let mut index = 0;
         let mut colored_index = 0;
 
-        // FIXME: VisibleEntities is ignored
-        for mut transparent_phase in views.iter_mut() {
+        for (visible_entities, mut transparent_phase) in views.iter_mut() {
             let extracted_sprites = &mut extracted_sprites.sprites;
             let image_bind_groups = &mut *image_bind_groups;
 
+            visible_entities_map.clear();
```

Comment:
Consider using a `FixedBitset` instead of a `HashMap` here. A contains check using that should be faster than a `HashSet`. Instead of storing the full entity, convert the `u32` Entity ID into a `usize` index, and use that to set and check the `FixedBitset`.

Reply:
I agree that that suggestion should be an improvement to performance. I want to try to figure out if there is a way to restructure so that the set can be avoided.

@james7132 did you ever try using a fixed bit set for visible entities? I seem to remember that you did but I don’t remember.

Also, random thought that is not thought through and that may not be good but could be, is it worth storing a fixed bit set of views in which an entity is visible, per entity, instead of a fixed bit set of entities visible per-view? Maybe not as maybe the per-view set is small enough to stay in cache where the per-entity sets would be loaded in as part of iteration… I don’t know.

Reply:
VisibleEntities IMO shouldn't, though I haven't tried. It'd require an &Entities on the other end to resolve the Entity from the ID, which may or may not be desirable depending on the downstream use case.